### ISSUE TITLE
Service information should be optional

### ISSUE THREAD

**Issue by sciguy16** [Ayrx/nmap_xml_parser#7]
I've got an Nmap file where not all of the discovered ports have identified services. Two problems:
* The error message references `state` instead of `service`
* The code expects to find a `service`, but this is apparently optional

https://github.com/Ayrx/nmap_xml_parser/blob/3f1a33257cf10091c4fe97e0c4cede4c2d005bea/src/port.rs#L70-L71

Here's a relevant chunk of the XML file:
```xml
<host starttime="1601435740" endtime="1601443647">
	<status state="up" reason="user-set" reason_ttl="0"/>
	<address addr="redacted" addrtype="ipv4"/>
	<hostnames>
		<hostname name="redacted" type="PTR"/>
	</hostnames>
	<ports>
		<extraports state="closed" count="65531">
			<extrareasons reason="resets" count="65531"/>
		</extraports>
		<port protocol="tcp" portid="443">
			<state state="open" reason="syn-ack" reason_ttl="246"/>
			<service name="https" method="table" conf="3"/>
		</port>
		<port protocol="tcp" portid="64443">
			<state state="open" reason="syn-ack" reason_ttl="246"/>
		</port>
	</ports>
	<times srtt="58747" rttvar="9426" to="100000"/>
</host>
```

**Comment by Ayrx** [comment]
Hey, just a ping to note that this and #8 has not been ignored.

I'll make an update to fix this this week :)

**Comment by Ayrx** [comment]
Fixed in d1ac318370d833db3b4456e6da86d176cda19442
